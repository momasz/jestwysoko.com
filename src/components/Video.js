import React from "react";
import PropTypes from "prop-types";
import {
  useParams
} from "react-router-dom";

const Video = ({ width = 560, height = 315 }) => {
  let { videoId } = useParams();
  const src = `https://www.youtube.com/embed/${videoId}?controls=0`;
  const allow = 'accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture';
  return (
    <iframe
      width={width}
      height={height}
      src={src}
      frameBorder="0"
      allow={allow}
      allowFullScreen
    ></iframe>
  )
};

Video.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number
};

export default Video;
