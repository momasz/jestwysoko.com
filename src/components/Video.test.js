import React from 'react';
import { mount } from 'enzyme';
import {
  BrowserRouter as Router,
} from "react-router-dom";
import Video from './Video';

describe('Video component', () => {
  it('renders as expected', () => {
    const video = mount(
      <Router>
        <Video/>
      </Router>
    );
    expect(video).toMatchSnapshot();
  });
});
