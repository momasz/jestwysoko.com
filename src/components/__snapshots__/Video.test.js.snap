// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video component renders as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Video>
      <div
        className="movie-content"
      >
        <div
          className="movie-container open"
        >
          <div
            className="video-container"
          >
            <div
              className="video-element"
            >
              <div
                className="loader"
              />
              <div
                className="video-wrapper"
              >
                <div
                  className="video"
                >
                  <iframe
                    allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
                    allowFullScreen={true}
                    frameBorder="0"
                    height={315}
                    loading="lazy"
                    onLoad={[Function]}
                    src="https://www.youtube.com/embed/undefined?controls=0"
                    width={560}
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </Video>
  </Router>
</BrowserRouter>
`;
